test/onnx/test_op_consistency.py

```diff
@@ -0,0 +1,462 @@
+)
+# fmt: on
+
+
+# END OF SECTION TO MODIFY #####################################################
+
+
+OPS_DB = copy.deepcopy(common_methods_invocations.op_db)
+
+
+class SingleOpModel(torch.nn.Module):
+    """Test model to wrap around a single op for export."""
+
+    def __init__(self, op, kwargs):
+        super().__init__()
+        self.operator = op
+        self.kwargs = kwargs
+
+    def forward(self, *args):
+        return self.operator(*args, **self.kwargs)
+
+
+class TestConsistency(common_utils.TestCase):
+    """Test consistency of exported ONNX models.
+
+    This is a parameterized test suite.
+    """
+
+    @classmethod
+    def create_test_base(cls, opset: int):
+        """Returns the base test method for the given opset."""
+
+        def _output_match_base(self, device: str, dtype: torch.dtype, op):
+            """Base test method for testing each opset."""
+            assert device == "cpu"
+
+            samples = op.sample_inputs(
+                device,
+                dtype,
+                requires_grad=False,
+            )
+
+            for (i, cpu_sample) in enumerate(samples):
+                # Provide the repr to subtest because tensors are not serializable in parallel test runs
+                with self.subTest(
+                    opset=opset,
+                    sample_num=i,
+                    input=repr(cpu_sample.input),
+                    args=repr(cpu_sample.args),
+                    kwargs=repr(cpu_sample.kwargs),
+                ):
+                    model = SingleOpModel(op, cpu_sample.kwargs)
+                    model.eval()
+
+                    # Run the test
+                    inputs = (cpu_sample.input, *cpu_sample.args)
+
+                    if dtype == torch.bfloat16:
+                        # Only export to ONNX without running with onnxruntime because
+                        # the CPU execution path for bfloat16 is not implemented in onnxruntime.
+                        model_buffer = io.BytesIO()
+                        torch.onnx.export(
+                            model, inputs, model_buffer, opset_version=opset
+                        )
+                        model_buffer.seek(0)
+                        onnx_model = onnx.load(model_buffer)
+                        onnx.checker.check_model(onnx_model, full_check=True)
+                        continue
+
+                    verification.verify(
+                        model,
+                        inputs,
+                        input_kwargs={},
+                        opset_version=opset,
+                        keep_initializers_as_inputs=True,
+                        ort_providers=ORT_PROVIDERS,
+                        check_shape=True,
+                        check_dtype=True,
+                        flatten=True,
+                    )
+
+        return _output_match_base
+
+    @classmethod
+    def parameterize_opsets(cls, opsets: Sequence[int]):
```

Comment:
I don't love this pattern but I guess it's OK. I think I would have done something like:

```
def _test_helper(...):
   ...

@ops(my_ops)
def test_opset11(...):
  if opset_skip(op, opset=11):
    unittest.skip(...)

  for op in ops:
    _test_helper(op, opset11_datatypes)

@ops(my_ops)
def test_opset12(...):
  ops = my_opset_filter(op_db, opset=12)
  for op in ops:
    _test_helper(op, opset12_datatypes)
```

Reply:
The only concern was I wanted to automatically enable new tests when the MAX_OPSET gets increased periodically. Do you have thoughts on how this can be done more cleanly?

Reply:
A pattern like this might be what you want:

```
opsets = (11, 12, ...)

for opset in opsets:
  test_name = make_test_name(opset)
  test_cls.test_name = partial(test_opset, opset)
```

But copying and pasting the tiny bit of wrapper code seems comparable to explaining the new opset, so I just wouldn't worry about it